### ISSUE TITLE
Problem: Login LibreTime

### ISSUE THREAD

**Issue by SGQuar** [LibreTime/libretime#410]
- Libretime ver 3.0.0-alpha.3
- OS: Ubutnu 14.04 LTS

Installed libretime successfully, all modules is working. Сan't login at system, gives error message ”Oops!!! Something went wrong!”

![sketch](https://user-images.githubusercontent.com/16132740/35065108-d7eed3c4-fbd4-11e7-95d3-3e59dd168bda.png)

Details in the log file **/var/log/airtime/zendphp.log**

2018-01-17T20:07:05+00:00 ERR (3): radio.pidos.eu [ErrorController.php:26 - errorAction()] - Could not parse version constraint ~tarball: Invalid version string "~tarball"
2018-01-17T20:07:05+00:00 ERR (3): radio.pidos.eu [ErrorController.php:27 - errorAction()] - #0 /usr/share/airtime/php/vendor/composer/semver/src/VersionParser.php(244): Composer\Semver\VersionParser->parseConstraint('~tarball')
#1 /usr/share/airtime/php/vendor/composer/semver/src/Semver.php(40): Composer\Semver\VersionParser->parseConstraints('~tarball instal...')
#2 /usr/share/airtime/php/vendor/composer/semver/src/Semver.php(56): Composer\Semver\Semver::satisfies('3.0.0-alpha.3', '~tarball instal...')
#3 [internal function]: Composer\Semver\Semver::Composer\Semver\{closure}('3.0.0-alpha.3')
#4 /usr/share/airtime/php/vendor/composer/semver/src/Semver.php(57): array_filter(Array, Object(Closure))
#5 /usr/share/airtime/php/airtime_mvc/application/views/helpers/VersionNotify.php(38): Composer\Semver\Semver::satisfiedBy(Array, '~tarball instal...')
#6 [internal function]: Airtime_View_Helper_VersionNotify->versionNotify()
#7 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/View/Abstract.php(350): call_user_func_array(Array, Array)
#8 /usr/share/airtime/php/airtime_mvc/application/layouts/scripts/layout.phtml(32): Zend_View_Abstract->__call('versionNotify', Array)
#9 /usr/share/airtime/php/airtime_mvc/application/layouts/scripts/layout.phtml(32): Zend_View->versionNotify()
#10 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/View.php(157): include('/usr/share/airt...')
#11 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/View/Abstract.php(888): Zend_View->_run('/usr/share/airt...')
#12 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Layout.php(796): Zend_View_Abstract->render('layout.phtml')
#13 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Layout/Controller/Plugin/Layout.php(143): Zend_Layout->render()
#14 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Plugin/Broker.php(333): Zend_Layout_Controller_Plugin_Layout->postDispatch(Object(Zend_Controller_Request_Http))
#15 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Front.php(965): Zend_Controller_Plugin_Broker->postDispatch(Object(Zend_Controller_Request_Http))
#16 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#17 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Application.php(384): Zend_Application_Bootstrap_Bootstrap->run()
#18 /usr/share/airtime/php/airtime_mvc/application/airtime-boot.php(84): Zend_Application->run()
#19 /usr/share/airtime/php/airtime_mvc/public/index.php(68): require_once('/usr/share/airt...')
#20 {main}
2018-01-17T20:07:05+00:00 ERR (3): radio.pidos.eu [ErrorController.php:54 - errorAction()] - An internal application error has occurred.: exception 'UnexpectedValueException' with message 'Could not parse version constraint ~tarball: Invalid version string 
"~tarball"' in /usr/share/airtime/php/vendor/composer/semver/src/VersionParser.php:485
Stack trace:
#0 /usr/share/airtime/php/vendor/composer/semver/src/VersionParser.php(244): Composer\Semver\VersionParser->parseConstraint('~tarball')
#1 /usr/share/airtime/php/vendor/composer/semver/src/Semver.php(40): Composer\Semver\VersionParser->parseConstraints('~tarball instal...')
#2 /usr/share/airtime/php/vendor/composer/semver/src/Semver.php(56): Composer\Semver\Semver::satisfies('3.0.0-alpha.3', '~tarball instal...')
#3 [internal function]: Composer\Semver\Semver::Composer\Semver\{closure}('3.0.0-alpha.3')
#4 /usr/share/airtime/php/vendor/composer/semver/src/Semver.php(57): array_filter(Array, Object(Closure))
#5 /usr/share/airtime/php/airtime_mvc/application/views/helpers/VersionNotify.php(38): Composer\Semver\Semver::satisfiedBy(Array, '~tarball instal...')
#6 [internal function]: Airtime_View_Helper_VersionNotify->versionNotify()
#7 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/View/Abstract.php(350): call_user_func_array(Array, Array)
#8 /usr/share/airtime/php/airtime_mvc/application/layouts/scripts/layout.phtml(32): Zend_View_Abstract->__call('versionNotify', Array)
#9 /usr/share/airtime/php/airtime_mvc/application/layouts/scripts/layout.phtml(32): Zend_View->versionNotify()
#10 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/View.php(157): include('/usr/share/airt...')
#11 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/View/Abstract.php(888): Zend_View->_run('/usr/share/airt...')
#12 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Layout.php(796): Zend_View_Abstract->render('layout.phtml')
#13 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Layout/Controller/Plugin/Layout.php(143): Zend_Layout->render()
#14 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Plugin/Broker.php(333): Zend_Layout_Controller_Plugin_Layout->postDispatch(Object(Zend_Controller_Request_Http))
#15 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Controller/Front.php(965): Zend_Controller_Plugin_Broker->postDispatch(Object(Zend_Controller_Request_Http))
#16 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#17 /usr/share/airtime/php/vendor/zendframework/zendframework1/library/Zend/Application.php(384): Zend_Application_Bootstrap_Bootstrap->run()
#18 /usr/share/airtime/php/airtime_mvc/application/airtime-boot.php(84): Zend_Application->run()
#19 /usr/share/airtime/php/airtime_mvc/public/index.php(68): require_once('/usr/share/airt...')
#20 {main}

**Comment by Robbt** [comment]
Ok, this seems like a relatively simple issue, but I'm not sure how to fix it. For some reason your version of LibreTime contains the string ~tarball install - this is causing the software to hiccup. I'll try to look it up to determine where it can be changed but you might try searching through your configuration for that string.

Ok, so some quick searching around github and on my codebase I found the line 
airtime_mvc/build/sql/defaultdata.sql:INSERT INTO cc_pref("keystr", "valstr") VALUES('schema_version', '3.0.0-alpha');

See if that string contains the ~tarball and possibly try to modify it and reinstall or modify the database value to remove the ~tarball portion of it.

The other question is why did your install happen to include this aspect and how can we prevent this from happening in the future. What distribution of linux are you installing LibreTime on ?

**Comment by SGQuar** [comment]
uname -a
Linux assus 3.13.0-24-generic #47-Ubuntu SMP Fri May 2 23:30:00 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

lsb_release -c
Codename:       trusty

**Comment by hairmare** [comment]
I'm pretty sure this is coming from the update checker. The tarball string was put into the VERSION file since we couldn't detect what version of LibreTime you got. This code is the culprit (I wrote it before I fixed the update checker and didn't test this scenario):

https://github.com/LibreTime/libretime/blob/dbf0a7be85ddcd75c70035ee91b16e9d28151bfc/build.sh#L19-L25

The workaround to fix this is to rewrite the VERSION file and rerun the installer using the same arguments you used the first time.

```bash
echo "3.0.0-alpha.3" > VERSION
./install # ...
```

The [releases](https://github.com/LibreTime/libretime/releases) contain a link to an enriched tarball that is complete with a properly generated VERSION file. If you just want to install LibreTime it's recommend that you use the enriched file. In this case it's called "libretime-3.0.0-alpha.3.tar.gz" and available from https://github.com/LibreTime/libretime/releases/download/3.0.0-alpha.3/libretime-3.0.0-alpha.3.tar.gz.

Thanks for the bug report. I'm looking into replacing the dirty basename hack in build.sh with something more sustainable. The hack in in there since there is no way to figure out what git tag you downloaded if you grab the non-enriched tarball.
